src/renderer_d3d12.cpp

```diff
@@ -5038,15 +5059,22 @@ namespace bgfx { namespace d3d12
 		const uint32_t rowPitch = layout.Footprint.RowPitch;
 
 		ID3D12Resource* staging = createCommittedResource(s_renderD3D12->m_device, HeapProperty::Upload, totalBytes);
-		uint8_t* data;
+		uint8_t* dstData;
 
 		D3D12_RANGE readRange = { 0, 0 };
-		DX_CHECK(staging->Map(0, &readRange, (void**)&data) );
-		for (uint32_t ii = 0, height = _rect.m_height; ii < height; ++ii)
+		DX_CHECK(staging->Map(0, &readRange, (void**)&dstData) );
+
+		for (uint32_t ii = 0, height = numRows; ii < height; ++ii)
```

Comment:
`numRows` instead of `_rect.m_height` because its more friendly for compression formats.